presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1321,6 +1324,48 @@ public void testDecimalComparedToWiderDecimal()
                 Range.lessThan(createDecimalType(4, 2), shortDecimal("99.99")), Range.greaterThan(createDecimalType(4, 2), shortDecimal("99.99"))), true));
     }
 
+    @Test
+    public void testVarcharComparedToCharExpression()
+            throws Exception
+    {
+        // greater than or equal
+        testSimpleComparison(greaterThanOrEqual(P, stringLiteral("123456789")), P, Range.greaterThan(createCharType(10), Slices.utf8Slice("12345678")));
```

Comment:
> Please use "abcd..." instead of "12345...." for test literals.

why `"12345...."` is disturbing? It makes it easier to write `CHAR` tests since we need to count trailing spaces. I prefer to leave it as is.

> Btw. Is varchar(x) coercible to char(x). If not this test does not make sense. Right?
> Also do we want to have test for each of such type paris. This is handled by generic mechanism.

Saturated casts are used to compute domain of input column which we coerce from. `CHAR` is coercible to `VARCHAR`. Therefore we need saturated cast from `VARCHAR` to `CHAR`.

> Probably instead making this class bloated we should test if SATURATED_FLOOR_CAST operator behaves correctly.

You are probably right. Currently there are many set of tests for each pair. I think saturated cast operators should be directly tested instead. I don't want to refactor it in this patch though. If you want I can remove `CHAR` tuple domain tests, but leaving them shouldn't hurt too.